src/plugins/expressions/common/expression_functions/specs/math.ts

```diff
@@ -130,10 +130,11 @@ export const math: ExpressionFunctionDefinition<
       throw errors.emptyExpression();
     }
 
+    // Use unique ID if available, otherwise fall back to names
     const mathContext = isDatatable(input)
       ? pivotObjectArray(
           input.rows,
-          input.columns.map((col) => col.name)
+          input.columns.map((col) => col.id)
```

Comment:
what is meant by fallback to names ? the fact that name will be used for columnId when id was not provided (to map_column for example ?)

Reply:
Yes, if no `id` is provided we find the first column with the name. This is required for backwards compatibility.